arch/arm/soc/atmel_sam0/samd/dts.fixup

```diff
@@ -0,0 +1,6 @@
+/* SoC level DTS fixup file */
+
+#define FLASH_DEV_BASE_ADDRESS		ATMEL_SAM0_NVM_CTRL_41004000_BASE_ADDRESS_0
```

Comment:
What will this be used for?  I ask as the flash function is broken into two pieces on the sam0:

- The nvmctrl, a peripheral at 0x41004000, which has the status and locking registers
- The flash itself, at addresses 0x0+, which you actually write to to do a write or erase

So, for example, a write would be:
- Write the page unlock to nvmctrl
- Write the data to the target address in flash itself
- Poll nvmctrl for the write to finish

Reply:
I put the base address there since I had it for the others, we can probably remove it if you'd like.